# HG changeset patch
# User J. Ryan Stinnett <jryans@gmail.com>
# Date 1441933077 18000
#      Thu Sep 10 19:57:57 2015 -0500
# Node ID 5767d57694e01d0a6255156804953a201de9ddb7
# Parent  f76e627e494e484d8e71ef8beb92c09d3e128e1e
Bug 912121 - Only one JS modules section per moz.build. rs=devtools

In a following patch, all DevTools moz.build files will use DevToolsModules to
install JS modules at a path that corresponds directly to their source tree
location.  Prior to this step, we must ensure each moz.build only defines a
single block of JS modules for easier automatic conversion.

diff --git a/devtools/client/debugger/content/moz.build b/devtools/client/debugger/content/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/debugger/content/moz.build
@@ -0,0 +1,15 @@
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+DIRS += [
+    'stores',
+    'views',
+]
+
+EXTRA_JS_MODULES.devtools.debugger.content += [
+    'constants.js',
+    'globalActions.js',
+    'utils.js'
+]
diff --git a/devtools/client/debugger/content/stores/moz.build b/devtools/client/debugger/content/stores/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/debugger/content/stores/moz.build
@@ -0,0 +1,9 @@
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.debugger.content.stores += [
+    'event-listeners.js',
+    'index.js'
+]
diff --git a/devtools/client/debugger/content/views/moz.build b/devtools/client/debugger/content/views/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/debugger/content/views/moz.build
@@ -0,0 +1,8 @@
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.debugger.content.views += [
+    'event-listeners-view.js'
+]
diff --git a/devtools/client/debugger/moz.build b/devtools/client/debugger/moz.build
--- a/devtools/client/debugger/moz.build
+++ b/devtools/client/debugger/moz.build
@@ -1,26 +1,15 @@
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+DIRS += [
+    'content',
+]
+
 EXTRA_JS_MODULES.devtools.debugger += [
     'debugger-commands.js',
     'panel.js'
 ]
 
-EXTRA_JS_MODULES.devtools.debugger.content += [
-    'content/constants.js',
-    'content/globalActions.js',
-    'content/utils.js'
-]
-
-EXTRA_JS_MODULES.devtools.debugger.content.views += [
-    'content/views/event-listeners-view.js'
-]
-
-EXTRA_JS_MODULES.devtools.debugger.content.stores += [
-    'content/stores/event-listeners.js',
-    'content/stores/index.js'
-]
-
 BROWSER_CHROME_MANIFESTS += ['test/mochitest/browser.ini']
diff --git a/devtools/client/framework/moz.build b/devtools/client/framework/moz.build
--- a/devtools/client/framework/moz.build
+++ b/devtools/client/framework/moz.build
@@ -1,22 +1,19 @@
 # -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 BROWSER_CHROME_MANIFESTS += ['test/browser.ini']
 
-EXTRA_JS_MODULES.devtools += [
+EXTRA_JS_MODULES.devtools.framework += [
     'gDevTools.jsm',
-    'ToolboxProcess.jsm',
-]
-
-EXTRA_JS_MODULES.devtools.framework += [
     'selection.js',
     'sidebar.js',
     'target.js',
     'toolbox-highlighter-utils.js',
     'toolbox-hosts.js',
     'toolbox-options.js',
     'toolbox.js',
+    'ToolboxProcess.jsm',
 ]
diff --git a/devtools/client/memory/modules/moz.build b/devtools/client/memory/modules/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/memory/modules/moz.build
@@ -0,0 +1,8 @@
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.memory += [
+    'census-view.js',
+]
diff --git a/devtools/client/memory/moz.build b/devtools/client/memory/moz.build
--- a/devtools/client/memory/moz.build
+++ b/devtools/client/memory/moz.build
@@ -1,11 +1,14 @@
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+DIRS += [
+    'modules',
+]
+
 EXTRA_JS_MODULES.devtools.memory += [
-    'modules/census-view.js',
     'panel.js',
 ]
 
 MOCHITEST_CHROME_MANIFESTS += ['test/mochitest/chrome.ini']
diff --git a/devtools/client/performance/modules/logic/moz.build b/devtools/client/performance/modules/logic/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/performance/modules/logic/moz.build
@@ -0,0 +1,12 @@
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.performance += [
+    'frame-utils.js',
+    'jit.js',
+    'marker-utils.js',
+    'tree-model.js',
+    'waterfall-utils.js',
+]
diff --git a/devtools/client/performance/modules/moz.build b/devtools/client/performance/modules/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/performance/modules/moz.build
@@ -0,0 +1,14 @@
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+DIRS += [
+    'logic',
+    'widgets',
+]
+
+EXTRA_JS_MODULES.devtools.performance += [
+    'global.js',
+    'markers.js',
+]
diff --git a/devtools/client/performance/modules/widgets/moz.build b/devtools/client/performance/modules/widgets/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/performance/modules/widgets/moz.build
@@ -0,0 +1,13 @@
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.performance += [
+    'graphs.js',
+    'marker-details.js',
+    'marker-view.js',
+    'markers-overview.js',
+    'tree-view.js',
+    'waterfall-ticks.js',
+]
diff --git a/devtools/client/performance/moz.build b/devtools/client/performance/moz.build
--- a/devtools/client/performance/moz.build
+++ b/devtools/client/performance/moz.build
@@ -1,24 +1,15 @@
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+DIRS += [
+    'modules',
+]
+
 EXTRA_JS_MODULES.devtools.performance += [
-    'modules/global.js',
-    'modules/logic/frame-utils.js',
-    'modules/logic/jit.js',
-    'modules/logic/marker-utils.js',
-    'modules/logic/tree-model.js',
-    'modules/logic/waterfall-utils.js',
-    'modules/markers.js',
-    'modules/widgets/graphs.js',
-    'modules/widgets/marker-details.js',
-    'modules/widgets/marker-view.js',
-    'modules/widgets/markers-overview.js',
-    'modules/widgets/tree-view.js',
-    'modules/widgets/waterfall-ticks.js',
-    'panel.js'
+    'panel.js',
 ]
 
 BROWSER_CHROME_MANIFESTS += ['test/browser.ini']
 XPCSHELL_TESTS_MANIFESTS += ['test/unit/xpcshell.ini']
diff --git a/devtools/client/projecteditor/lib/helpers/moz.build b/devtools/client/projecteditor/lib/helpers/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/helpers/moz.build
@@ -0,0 +1,13 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.helpers += [
+    'event.js',
+    'file-picker.js',
+    'l10n.js',
+    'prompts.js',
+    'readdir.js',
+]
diff --git a/devtools/client/projecteditor/lib/moz.build b/devtools/client/projecteditor/lib/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/moz.build
@@ -0,0 +1,19 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+DIRS += [
+    'helpers',
+    'plugins',
+    'stores',
+]
+
+EXTRA_JS_MODULES.devtools.projecteditor += [
+    'editors.js',
+    'project.js',
+    'projecteditor.js',
+    'shells.js',
+    'tree.js',
+]
diff --git a/devtools/client/projecteditor/lib/plugins/app-manager/moz.build b/devtools/client/projecteditor/lib/plugins/app-manager/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/plugins/app-manager/moz.build
@@ -0,0 +1,10 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.plugins['app-manager'] += [
+    'app-project-editor.js',
+    'plugin.js',
+]
diff --git a/devtools/client/projecteditor/lib/plugins/delete/moz.build b/devtools/client/projecteditor/lib/plugins/delete/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/plugins/delete/moz.build
@@ -0,0 +1,9 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.plugins.delete += [
+    'delete.js',
+]
diff --git a/devtools/client/projecteditor/lib/plugins/dirty/moz.build b/devtools/client/projecteditor/lib/plugins/dirty/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/plugins/dirty/moz.build
@@ -0,0 +1,9 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.plugins.dirty += [
+    'dirty.js',
+]
diff --git a/devtools/client/projecteditor/lib/plugins/image-view/moz.build b/devtools/client/projecteditor/lib/plugins/image-view/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/plugins/image-view/moz.build
@@ -0,0 +1,10 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.plugins['image-view'] += [
+    'image-editor.js',
+    'plugin.js',
+]
diff --git a/devtools/client/projecteditor/lib/plugins/logging/moz.build b/devtools/client/projecteditor/lib/plugins/logging/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/plugins/logging/moz.build
@@ -0,0 +1,9 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.plugins.logging += [
+    'logging.js',
+]
diff --git a/devtools/client/projecteditor/lib/plugins/moz.build b/devtools/client/projecteditor/lib/plugins/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/plugins/moz.build
@@ -0,0 +1,21 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+DIRS += [
+    'app-manager',
+    'delete',
+    'dirty',
+    'image-view',
+    'logging',
+    'new',
+    'rename',
+    'save',
+    'status-bar',
+]
+
+EXTRA_JS_MODULES.devtools.projecteditor.plugins += [
+    'core.js',
+]
diff --git a/devtools/client/projecteditor/lib/plugins/new/moz.build b/devtools/client/projecteditor/lib/plugins/new/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/plugins/new/moz.build
@@ -0,0 +1,9 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.plugins.new += [
+    'new.js',
+]
diff --git a/devtools/client/projecteditor/lib/plugins/rename/moz.build b/devtools/client/projecteditor/lib/plugins/rename/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/plugins/rename/moz.build
@@ -0,0 +1,9 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.plugins.rename += [
+    'rename.js',
+]
diff --git a/devtools/client/projecteditor/lib/plugins/save/moz.build b/devtools/client/projecteditor/lib/plugins/save/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/plugins/save/moz.build
@@ -0,0 +1,9 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.plugins.save += [
+    'save.js',
+]
diff --git a/devtools/client/projecteditor/lib/plugins/status-bar/moz.build b/devtools/client/projecteditor/lib/plugins/status-bar/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/plugins/status-bar/moz.build
@@ -0,0 +1,9 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.plugins['status-bar'] += [
+    'plugin.js',
+]
diff --git a/devtools/client/projecteditor/lib/stores/moz.build b/devtools/client/projecteditor/lib/stores/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/projecteditor/lib/stores/moz.build
@@ -0,0 +1,11 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.projecteditor.stores += [
+    'base.js',
+    'local.js',
+    'resource.js',
+]
diff --git a/devtools/client/projecteditor/moz.build b/devtools/client/projecteditor/moz.build
--- a/devtools/client/projecteditor/moz.build
+++ b/devtools/client/projecteditor/moz.build
@@ -1,70 +1,9 @@
 # -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+DIRS += ['lib']
+
 BROWSER_CHROME_MANIFESTS += ['test/browser.ini']
-
-EXTRA_JS_MODULES.devtools.projecteditor += [
-    'lib/editors.js',
-    'lib/project.js',
-    'lib/projecteditor.js',
-    'lib/shells.js',
-    'lib/tree.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.helpers += [
-    'lib/helpers/event.js',
-    'lib/helpers/file-picker.js',
-    'lib/helpers/l10n.js',
-    'lib/helpers/prompts.js',
-    'lib/helpers/readdir.js',
-]
-EXTRA_JS_MODULES.devtools.projecteditor.plugins += [
-    'lib/plugins/core.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.plugins['app-manager'] += [
-    'lib/plugins/app-manager/app-project-editor.js',
-    'lib/plugins/app-manager/plugin.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.plugins.delete += [
-    'lib/plugins/delete/delete.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.plugins.dirty += [
-    'lib/plugins/dirty/dirty.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.plugins['image-view'] += [
-    'lib/plugins/image-view/image-editor.js',
-    'lib/plugins/image-view/plugin.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.plugins.logging += [
-    'lib/plugins/logging/logging.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.plugins.new += [
-    'lib/plugins/new/new.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.plugins.rename += [
-    'lib/plugins/rename/rename.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.plugins.save += [
-    'lib/plugins/save/save.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.plugins['status-bar'] += [
-    'lib/plugins/status-bar/plugin.js',
-]
-
-EXTRA_JS_MODULES.devtools.projecteditor.stores += [
-    'lib/stores/base.js',
-    'lib/stores/local.js',
-    'lib/stores/resource.js',
-]
diff --git a/devtools/client/responsivedesign/moz.build b/devtools/client/responsivedesign/moz.build
--- a/devtools/client/responsivedesign/moz.build
+++ b/devtools/client/responsivedesign/moz.build
@@ -1,14 +1,11 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+BROWSER_CHROME_MANIFESTS += ['test/browser.ini']
+
 EXTRA_JS_MODULES.devtools += [
+    'resize-commands.js',
     'responsivedesign-child.js',
-    'responsivedesign.jsm'
+    'responsivedesign.jsm',
 ]
-EXTRA_JS_MODULES.devtools.responsivedesign += [
-    'resize-commands.js'
-]
-
-
-BROWSER_CHROME_MANIFESTS += ['test/browser.ini']
diff --git a/devtools/client/scratchpad/moz.build b/devtools/client/scratchpad/moz.build
--- a/devtools/client/scratchpad/moz.build
+++ b/devtools/client/scratchpad/moz.build
@@ -1,16 +1,13 @@
 # -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-EXTRA_JS_MODULES.devtools += [
-    'scratchpad-manager.jsm',
-]
-
 EXTRA_JS_MODULES.devtools.scratchpad += [
     'scratchpad-commands.js',
+    'scratchpad-manager.jsm',
     'scratchpad-panel.js',
 ]
 
 BROWSER_CHROME_MANIFESTS += ['test/browser.ini']
diff --git a/devtools/client/shared/content/moz.build b/devtools/client/shared/content/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/shared/content/moz.build
@@ -0,0 +1,14 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.shared.content += [
+    'react.js'
+]
+
+if CONFIG['DEBUG_JS_MODULES']:
+    EXTRA_JS_MODULES.devtools.shared.content += [
+        'react-dev.js'
+    ]
diff --git a/devtools/client/shared/moz.build b/devtools/client/shared/moz.build
--- a/devtools/client/shared/moz.build
+++ b/devtools/client/shared/moz.build
@@ -1,78 +1,41 @@
 # -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+DIRS += [
+    'content',
+    'fluxify',
+    'widgets',
+]
+
 BROWSER_CHROME_MANIFESTS += ['test/browser.ini']
 XPCSHELL_TESTS_MANIFESTS += ['test/unit/xpcshell.ini']
 
-EXTRA_JS_MODULES.devtools += [
+EXTRA_JS_MODULES.devtools.shared += [
     'AppCacheUtils.jsm',
-    'Curl.jsm',
-    'DeveloperToolbar.jsm',
-    'DOMHelpers.jsm',
-    'Jsbeautify.jsm',
-    'Parser.jsm',
-    'SplitView.jsm',
-]
-
-EXTRA_JS_MODULES.devtools += [
-    'widgets/AbstractTreeItem.jsm',
-    'widgets/BreadcrumbsWidget.jsm',
-    'widgets/Chart.jsm',
-    'widgets/GraphsWorker.js',
-    'widgets/SideMenuWidget.jsm',
-    'widgets/SimpleListWidget.jsm',
-    'widgets/VariablesView.jsm',
-    'widgets/VariablesViewController.jsm',
-    'widgets/ViewHelpers.jsm',
-]
-
-EXTRA_JS_MODULES.devtools.shared += [
     'autocomplete-popup.js',
     'browser-loader.js',
     'create-dispatcher.js',
+    'Curl.jsm',
+    'DeveloperToolbar.jsm',
     'devices.js',
+    'DOMHelpers.jsm',
     'doorhanger.js',
     'frame-script-utils.js',
     'getjson.js',
     'inplace-editor.js',
+    'Jsbeautify.jsm',
     'node-attribute-parser.js',
     'observable-object.js',
     'options-view.js',
+    'Parser.jsm',
     'poller.js',
     'source-utils.js',
+    'SplitView.jsm',
     'telemetry.js',
     'theme-switching.js',
     'theme.js',
     'undo.js'
 ]
-
-EXTRA_JS_MODULES.devtools.shared.content += [
-    'content/react.js'
-]
-
-if CONFIG['DEBUG_JS_MODULES']:
-    EXTRA_JS_MODULES.devtools.shared.content += [
-        'content/react-dev.js'
-    ]
-
-EXTRA_JS_MODULES.devtools.shared.widgets += [
-    'widgets/BarGraphWidget.js',
-    'widgets/CubicBezierPresets.js',
-    'widgets/CubicBezierWidget.js',
-    'widgets/FastListWidget.js',
-    'widgets/FilterWidget.js',
-    'widgets/FlameGraph.js',
-    'widgets/Graphs.js',
-    'widgets/LineGraphWidget.js',
-    'widgets/MdnDocsWidget.js',
-    'widgets/MountainGraphWidget.js',
-    'widgets/Spectrum.js',
-    'widgets/TableWidget.js',
-    'widgets/Tooltip.js',
-    'widgets/TreeWidget.js',
-]
-
-DIRS += ['fluxify']
diff --git a/devtools/client/shared/widgets/moz.build b/devtools/client/shared/widgets/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/shared/widgets/moz.build
@@ -0,0 +1,31 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.shared.widgets += [
+    'AbstractTreeItem.jsm',
+    'BarGraphWidget.js',
+    'BreadcrumbsWidget.jsm',
+    'Chart.jsm',
+    'CubicBezierPresets.js',
+    'CubicBezierWidget.js',
+    'FastListWidget.js',
+    'FilterWidget.js',
+    'FlameGraph.js',
+    'Graphs.js',
+    'GraphsWorker.js',
+    'LineGraphWidget.js',
+    'MdnDocsWidget.js',
+    'MountainGraphWidget.js',
+    'SideMenuWidget.jsm',
+    'SimpleListWidget.jsm',
+    'Spectrum.js',
+    'TableWidget.js',
+    'Tooltip.js',
+    'TreeWidget.js',
+    'VariablesView.jsm',
+    'VariablesViewController.jsm',
+    'ViewHelpers.jsm',
+]
diff --git a/devtools/client/styleeditor/moz.build b/devtools/client/styleeditor/moz.build
--- a/devtools/client/styleeditor/moz.build
+++ b/devtools/client/styleeditor/moz.build
@@ -1,19 +1,16 @@
 # -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 BROWSER_CHROME_MANIFESTS += ['test/browser.ini']
 
-EXTRA_JS_MODULES.devtools += [
+EXTRA_JS_MODULES.devtools.styleeditor += [
+    'styleeditor-commands.js',
+    'styleeditor-panel.js',
     'StyleEditorUI.jsm',
     'StyleEditorUtil.jsm',
     'StyleSheetEditor.jsm',
-]
-
-EXTRA_JS_MODULES.devtools.styleeditor += [
-    'styleeditor-commands.js',
-    'styleeditor-panel.js',
     'utils.js',
 ]
diff --git a/devtools/client/webide/modules/moz.build b/devtools/client/webide/modules/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/client/webide/modules/moz.build
@@ -0,0 +1,19 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.webide += [
+    'addons.js',
+    'app-manager.js',
+    'build.js',
+    'config-view.js',
+    'project-list.js',
+    'runtime-list.js',
+    'runtimes.js',
+    'simulator-process.js',
+    'simulators.js',
+    'tab-store.js',
+    'utils.js'
+]
diff --git a/devtools/client/webide/moz.build b/devtools/client/webide/moz.build
--- a/devtools/client/webide/moz.build
+++ b/devtools/client/webide/moz.build
@@ -2,38 +2,24 @@
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 DIRS += [
     'content',
     'components',
+    'modules',
     'themes',
 ]
 
 BROWSER_CHROME_MANIFESTS += [
     'test/browser.ini',
     'test/sidebars/browser.ini'
 ]
 MOCHITEST_CHROME_MANIFESTS += [
     'test/chrome.ini',
     'test/sidebars/chrome.ini'
 ]
 
-EXTRA_JS_MODULES.devtools.webide += [
-    'modules/addons.js',
-    'modules/app-manager.js',
-    'modules/build.js',
-    'modules/config-view.js',
-    'modules/project-list.js',
-    'modules/runtime-list.js',
-    'modules/runtimes.js',
-    'modules/simulator-process.js',
-    'modules/simulators.js',
-    'modules/tab-store.js',
-    'modules/utils.js'
-]
-
 JS_PREFERENCE_FILES += [
     'webide-prefs.js',
 ]
-
diff --git a/devtools/server/actors/moz.build b/devtools/server/actors/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/server/actors/moz.build
@@ -0,0 +1,57 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+DIRS += [
+    'utils',
+]
+
+EXTRA_JS_MODULES.devtools.server.actors += [
+    'actor-registry.js',
+    'addon.js',
+    'animation.js',
+    'call-watcher.js',
+    'canvas.js',
+    'child-process.js',
+    'childtab.js',
+    'chrome.js',
+    'common.js',
+    'csscoverage.js',
+    'device.js',
+    'director-manager.js',
+    'director-registry.js',
+    'eventlooplag.js',
+    'framerate.js',
+    'gcli.js',
+    'highlighter.css',
+    'highlighter.js',
+    'inspector.js',
+    'layout.js',
+    'memory.js',
+    'monitor.js',
+    'object.js',
+    'performance-entries.js',
+    'performance-recording.js',
+    'performance.js',
+    'preference.js',
+    'pretty-print-worker.js',
+    'profiler.js',
+    'promises.js',
+    'root.js',
+    'script.js',
+    'settings.js',
+    'storage.js',
+    'string.js',
+    'styleeditor.js',
+    'styles.js',
+    'stylesheets.js',
+    'timeline.js',
+    'webapps.js',
+    'webaudio.js',
+    'webbrowser.js',
+    'webconsole.js',
+    'webgl.js',
+    'worker.js',
+]
diff --git a/devtools/server/actors/utils/moz.build b/devtools/server/actors/utils/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/server/actors/utils/moz.build
@@ -0,0 +1,16 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.server.actors.utils += [
+    'actor-registry-utils.js',
+    'audionodes.json',
+    'automation-timeline.js',
+    'make-debugger.js',
+    'map-uri-to-addon-id.js',
+    'ScriptStore.js',
+    'stack.js',
+    'TabSources.js'
+]
diff --git a/devtools/server/moz.build b/devtools/server/moz.build
--- a/devtools/server/moz.build
+++ b/devtools/server/moz.build
@@ -1,95 +1,37 @@
 # -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 include('../templates.mozbuild')
 
+DIRS += [
+    'actors',
+]
+
 BROWSER_CHROME_MANIFESTS += ['tests/browser/browser.ini']
 MOCHITEST_CHROME_MANIFESTS += ['tests/mochitest/chrome.ini']
 XPCSHELL_TESTS_MANIFESTS += ['tests/unit/xpcshell.ini']
 
 XPIDL_SOURCES += [
     'nsIJSInspector.idl',
 ]
 
 XPIDL_MODULE = 'jsinspector'
 
 SOURCES += [
     'nsJSInspector.cpp',
 ]
 
 FINAL_LIBRARY = 'xul'
 
-EXTRA_JS_MODULES.devtools += [
-    'content-server.jsm',
-    'dbg-server.jsm',
-]
-
 EXTRA_JS_MODULES.devtools.server += [
     'child.js',
     'content-globals.js',
+    'content-server.jsm',
+    'dbg-server.jsm',
     'main.js',
     'protocol.js',
     'worker.js'
 ]
-
-EXTRA_JS_MODULES.devtools.server.actors += [
-    'actors/actor-registry.js',
-    'actors/addon.js',
-    'actors/animation.js',
-    'actors/call-watcher.js',
-    'actors/canvas.js',
-    'actors/child-process.js',
-    'actors/childtab.js',
-    'actors/chrome.js',
-    'actors/common.js',
-    'actors/csscoverage.js',
-    'actors/device.js',
-    'actors/director-manager.js',
-    'actors/director-registry.js',
-    'actors/eventlooplag.js',
-    'actors/framerate.js',
-    'actors/gcli.js',
-    'actors/highlighter.css',
-    'actors/highlighter.js',
-    'actors/inspector.js',
-    'actors/layout.js',
-    'actors/memory.js',
-    'actors/monitor.js',
-    'actors/object.js',
-    'actors/performance-entries.js',
-    'actors/performance-recording.js',
-    'actors/performance.js',
-    'actors/preference.js',
-    'actors/pretty-print-worker.js',
-    'actors/profiler.js',
-    'actors/promises.js',
-    'actors/root.js',
-    'actors/script.js',
-    'actors/settings.js',
-    'actors/storage.js',
-    'actors/string.js',
-    'actors/styleeditor.js',
-    'actors/styles.js',
-    'actors/stylesheets.js',
-    'actors/timeline.js',
-    'actors/webapps.js',
-    'actors/webaudio.js',
-    'actors/webbrowser.js',
-    'actors/webconsole.js',
-    'actors/webgl.js',
-    'actors/worker.js',
-]
-
-EXTRA_JS_MODULES.devtools.server.actors.utils += [
-    'actors/utils/actor-registry-utils.js',
-    'actors/utils/audionodes.json',
-    'actors/utils/automation-timeline.js',
-    'actors/utils/make-debugger.js',
-    'actors/utils/map-uri-to-addon-id.js',
-    'actors/utils/ScriptStore.js',
-    'actors/utils/stack.js',
-    'actors/utils/TabSources.js'
-]
diff --git a/devtools/shared/client/moz.build b/devtools/shared/client/moz.build
--- a/devtools/shared/client/moz.build
+++ b/devtools/shared/client/moz.build
@@ -1,14 +1,11 @@
 # -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-EXTRA_JS_MODULES.devtools += [
-    'dbg-client.jsm',
-]
-
 EXTRA_JS_MODULES.devtools.client += [
     'connection-manager.js',
+    'dbg-client.jsm',
     'main.js',
 ]
diff --git a/devtools/shared/gcli/commands/moz.build b/devtools/shared/gcli/commands/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/shared/gcli/commands/moz.build
@@ -0,0 +1,28 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.gcli.commands += [
+    'addon.js',
+    'appcache.js',
+    'calllog.js',
+    'cmd.js',
+    'cookie.js',
+    'csscoverage.js',
+    'folder.js',
+    'highlight.js',
+    'index.js',
+    'inject.js',
+    'jsb.js',
+    'listen.js',
+    'media.js',
+    'pagemod.js',
+    'paintflashing.js',
+    'restart.js',
+    'rulers.js',
+    'screenshot.js',
+    'security.js',
+    'tools.js',
+]
diff --git a/devtools/shared/gcli/moz.build b/devtools/shared/gcli/moz.build
--- a/devtools/shared/gcli/moz.build
+++ b/devtools/shared/gcli/moz.build
@@ -1,35 +1,16 @@
 # -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-EXTRA_JS_MODULES.devtools.gcli.commands += [
-    'commands/addon.js',
-    'commands/appcache.js',
-    'commands/calllog.js',
-    'commands/cmd.js',
-    'commands/cookie.js',
-    'commands/csscoverage.js',
-    'commands/folder.js',
-    'commands/highlight.js',
-    'commands/index.js',
-    'commands/inject.js',
-    'commands/jsb.js',
-    'commands/listen.js',
-    'commands/media.js',
-    'commands/pagemod.js',
-    'commands/paintflashing.js',
-    'commands/restart.js',
-    'commands/rulers.js',
-    'commands/screenshot.js',
-    'commands/security.js',
-    'commands/tools.js',
+DIRS += [
+    'commands',
 ]
 
 EXTRA_JS_MODULES.devtools.gcli.commands += [
     'source/lib/gcli/commands/clear.js',
     'source/lib/gcli/commands/commands.js',
     'source/lib/gcli/commands/connect.js',
     'source/lib/gcli/commands/context.js',
     'source/lib/gcli/commands/exec.js',
diff --git a/devtools/shared/jsbeautify/lib/moz.build b/devtools/shared/jsbeautify/lib/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/shared/jsbeautify/lib/moz.build
@@ -0,0 +1,10 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.jsbeautify += [
+    'sanitytest.js',
+    'urlencode_unpacker.js',
+]
diff --git a/devtools/shared/jsbeautify/moz.build b/devtools/shared/jsbeautify/moz.build
--- a/devtools/shared/jsbeautify/moz.build
+++ b/devtools/shared/jsbeautify/moz.build
@@ -1,17 +1,16 @@
 # -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+DIRS += [
+    'lib',
+    'src',
+]
+
 XPCSHELL_TESTS_MANIFESTS += ['tests/unit/xpcshell.ini']
 
 EXTRA_JS_MODULES.devtools.jsbeautify += [
     'beautify.js',
-    'lib/sanitytest.js',
-    'lib/urlencode_unpacker.js',
-    'src/beautify-css.js',
-    'src/beautify-html.js',
-    'src/beautify-js.js',
-    'src/beautify-tests.js'
 ]
diff --git a/devtools/shared/jsbeautify/src/moz.build b/devtools/shared/jsbeautify/src/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/shared/jsbeautify/src/moz.build
@@ -0,0 +1,12 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.jsbeautify += [
+    'beautify-css.js',
+    'beautify-html.js',
+    'beautify-js.js',
+    'beautify-tests.js'
+]
diff --git a/devtools/shared/moz.build b/devtools/shared/moz.build
--- a/devtools/shared/moz.build
+++ b/devtools/shared/moz.build
@@ -27,24 +27,21 @@ DIRS += [
     'webconsole'
 ]
 
 MOCHITEST_CHROME_MANIFESTS += ['tests/mochitest/chrome.ini']
 XPCSHELL_TESTS_MANIFESTS += ['tests/unit/xpcshell.ini']
 
 EXTRA_JS_MODULES.devtools += [
     'async-utils.js',
+    'Console.jsm',
     'content-observer.js',
     'css-color.js',
     'deprecated-sync-thenables.js',
     'DevToolsUtils.js',
     'event-emitter.js',
     'event-parsers.js',
     'layout-helpers.js',
+    'Loader.jsm',
     'output-parser.js',
     'path.js',
     'worker-loader.js',
 ]
-
-EXTRA_JS_MODULES.devtools += [
-    'Console.jsm',
-    'Loader.jsm',
-]
diff --git a/devtools/shared/performance/legacy/moz.build b/devtools/shared/performance/legacy/moz.build
new file mode 100644
--- /dev/null
+++ b/devtools/shared/performance/legacy/moz.build
@@ -0,0 +1,12 @@
+# -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
+# vim: set filetype=python:
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+EXTRA_JS_MODULES.devtools.performance.legacy += [
+  'actors.js',
+  'compatibility.js',
+  'front.js',
+  'recording.js',
+]
diff --git a/devtools/shared/performance/moz.build b/devtools/shared/performance/moz.build
--- a/devtools/shared/performance/moz.build
+++ b/devtools/shared/performance/moz.build
@@ -1,22 +1,19 @@
 # -*- Mode: python; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 40 -*-
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+DIRS += [
+    'legacy',
+]
+
 XPCSHELL_TESTS_MANIFESTS += ['test/xpcshell.ini']
 
 EXTRA_JS_MODULES.devtools.performance += [
   'io.js',
   'process-communication.js',
   'recorder.js',
   'recording-common.js',
   'utils.js',
 ]
-
-EXTRA_JS_MODULES.devtools.performance.legacy += [
-  'legacy/actors.js',
-  'legacy/compatibility.js',
-  'legacy/front.js',
-  'legacy/recording.js',
-]
